From af303c09452259b8cf6c7bda0e101f40096b5ab5 Mon Sep 17 00:00:00 2001
From: Chengyi Zhao <zhaochengyi@uniontech.com>
Date: Thu, 5 Aug 2021 16:54:57 +0800
Subject: [PATCH] alsa-mixer: Add dahua-4f05-usb-audio.conf to adapt the usb
 audio card(USB0bda:4f05) which it is used in Dahua device.

---
 src/Makefile.am                               |  3 +-
 .../alsa/mixer/paths/analog-input-mic.conf    |  3 ++
 .../mixer/paths/analog-output-speaker.conf    |  6 +++
 .../mixer/profile-sets/90-pulseaudio.rules    |  2 +
 .../profile-sets/dahua-4f05-usb-audio.conf    | 46 +++++++++++++++++++
 5 files changed, 59 insertions(+), 1 deletion(-)
 create mode 100644 src/modules/alsa/mixer/profile-sets/dahua-4f05-usb-audio.conf

diff --git a/src/Makefile.am b/src/Makefile.am
index 9a6e42f..2866171 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -1325,7 +1325,8 @@ dist_alsaprofilesets_DATA = \
 		modules/alsa/mixer/profile-sets/st2557-pc-usb-audio.conf \
 		modules/alsa/mixer/profile-sets/22d4-us-usb-audio.conf \
 		modules/alsa/mixer/profile-sets/msi-disable-digital.conf \
-		modules/alsa/mixer/profile-sets/loongson-laptop-audio.conf
+		modules/alsa/mixer/profile-sets/loongson-laptop-audio.conf \
+		modules/alsa/mixer/profile-sets/dahua-4f05-usb-audio.conf
 
 if HAVE_UDEV
 dist_udevrules_DATA = \
diff --git a/src/modules/alsa/mixer/paths/analog-input-mic.conf b/src/modules/alsa/mixer/paths/analog-input-mic.conf
index 123439b..3cb3ced 100644
--- a/src/modules/alsa/mixer/paths/analog-input-mic.conf
+++ b/src/modules/alsa/mixer/paths/analog-input-mic.conf
@@ -24,6 +24,9 @@ description-key = analog-input-microphone
 [Jack Mic]
 required-any = any
 
+[Jack Mic - Input]
+required-any = any
+
 [Jack Mic Phantom]
 required-any = any
 state.plugged = unknown
diff --git a/src/modules/alsa/mixer/paths/analog-output-speaker.conf b/src/modules/alsa/mixer/paths/analog-output-speaker.conf
index 8b1115d..8ba54ed 100644
--- a/src/modules/alsa/mixer/paths/analog-output-speaker.conf
+++ b/src/modules/alsa/mixer/paths/analog-output-speaker.conf
@@ -24,6 +24,12 @@ description-key = analog-output-speaker
 [Properties]
 device.icon_name = audio-speakers
 
+[Jack Speaker - Output]
+required-any = any
+
+[Jack Speaker]
+required-any = any
+
 [Jack Headphone]
 state.plugged = no
 state.unplugged = unknown
diff --git a/src/modules/alsa/mixer/profile-sets/90-pulseaudio.rules b/src/modules/alsa/mixer/profile-sets/90-pulseaudio.rules
index 121af4e..a83c5f2 100644
--- a/src/modules/alsa/mixer/profile-sets/90-pulseaudio.rules
+++ b/src/modules/alsa/mixer/profile-sets/90-pulseaudio.rules
@@ -120,5 +120,7 @@ ATTRS{idVendor}=="1038", ATTRS{idProduct}=="1294", ENV{PULSE_PROFILE_SET}="steel
 ATTRS{idVendor}=="0bda", ATTRS{idProduct}=="4987", ENV{PULSE_PROFILE_SET}="st2557-pc-usb-audio.conf"
 # Sugon W330-H35A
 ATTRS{idVendor}=="0bda", ATTRS{idProduct}=="491b", ENV{PULSE_PROFILE_SET}="22d4-us-usb-audio.conf"
+# Dahua
+ATTRS{idVendor}=="0bda", ATTRS{idProduct}=="4f05", ENV{PULSE_PROFILE_SET}="dahua-4f05-usb-audio.conf"
 
 LABEL="pulseaudio_end"
diff --git a/src/modules/alsa/mixer/profile-sets/dahua-4f05-usb-audio.conf b/src/modules/alsa/mixer/profile-sets/dahua-4f05-usb-audio.conf
new file mode 100644
index 0000000..9aded93
--- /dev/null
+++ b/src/modules/alsa/mixer/profile-sets/dahua-4f05-usb-audio.conf
@@ -0,0 +1,46 @@
+# This file is part of PulseAudio.
+#
+# PulseAudio is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as
+# published by the Free Software Foundation; either version 2.1 of the
+# License, or (at your option) any later version.
+#
+# PulseAudio is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with PulseAudio; if not, see <http://www.gnu.org/licenses/>.
+
+[General]
+auto-profiles = yes
+
+[Mapping analog-stereo-mic]
+description = Micphone
+paths-input = analog-input analog-input-mic
+device-strings = hw:%f,2
+channel-map = left,right
+direction = input
+
+[Mapping analog-stereo-linein]
+description = Rear Linein
+paths-input = analog-input analog-input-linein
+device-strings = hw:%f,1
+channel-map = left,right
+direction = input
+
+[Mapping analog-stereo-headphone]
+description = Front Headphone
+paths-output = analog-output analog-output-headphones
+device-strings = hw:%f,1
+channel-map = left,right
+direction = output
+
+[Mapping analog-stereo-speaker]
+description = Rear Speaker
+paths-output = analog-output analog-output-speaker
+device-strings = hw:%f,0
+channel-map = left,right
+direction = output
+
-- 
2.20.1

